src/main/java/com/faforever/client/map/generator/MapGeneratorService.java

```diff
@@ -124,26 +152,62 @@ protected String queryNewestVersion() {
     JsonElement jsonElement = new JsonParser().parse(response.getBody());
     JsonObject mainObject = jsonElement.getAsJsonObject();
 
-    return mainObject.get("tag_name").getAsString();
+    return new ComparableVersion(mainObject.get("tag_name").getAsString());
+  }
+
+  @VisibleForTesting
+  @Cacheable(CacheNames.MAP_GENERATOR)
```

Comment:
I'm not entirely sure what happens if you use the cache twice for the same type as the method has no arguments?!

Reply:
Alright well I had left in the query newest version in case it could be used elsewhere but it is not used anywhere else so I removed it.